### PR TITLE
GetFeeds must always discard actions with dangling repo_id

### LINKED ISSUE #19631: NPE in routers/web/feed/profile.go:37 on 1.16.7 following failed migration
Extract from a comment in #19613 

```
2022/05/05 20:06:26 ...s/graceful/server.go:61:NewServer() [I] Starting new Web server: tcp:0.0.0.0:3000 on PID: 16
2022/05/05 20:06:46 Started GET / for 10.###.###.###:0
2022/05/05 20:06:46 routers/web/base.go:130:1() [E] PANIC: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:220 (0x457635)
/usr/local/go/src/runtime/signal_unix.go:818 (0x457605)
/go/src/code.gitea.io/gitea/routers/web/feed/profile.go:37 (0x1fbb59c)
/go/src/code.gitea.io/gitea/routers/web/user/home.go:134 (0x1fd20d2)
/go/src/code.gitea.io/gitea/routers/web/home.go:41 (0x2034770)
/go/src/code.gitea.io/gitea/modules/web/route.go:74 (0x1e47d44)
/usr/local/go/src/net/http/server.go:2084 (0x7ac6ce)
/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:442 (0x1656dd5)
/usr/local/go/src/net/http/server.go:2084 (0x7ac6ce)
...
```

The migration consists of pushing cloned repo's from GitHub into Gitea, the repo's are created using the API and data transferred using Git https.

Now that I have migrated a few more repo's, the 500 error on the landing page is gone. Presumably because whatever was causing it is not longer shown on it.

_Originally posted by @number42net in https://github.com/go-gitea/gitea/issues/19613#issuecomment-1119134431_

### PR DESCRIPTION
**Pull request by singuliere**
See original [merge request from Loïc Dachary @ forgefriends](https://lab.forgefriends.org/forgefriends/forgefriends/-/merge_requests/48)

---

See https://discourse.gitea.io/t/blank-page-after-login/5051/12
for a panic in 1.16.6.

Fix #19631

### CODE REVIEW

File: models/fixtures/action.yml [hunk 1]
```diff
@@ -56,3 +56,11 @@
   repo_id: 8 # public
   is_private: false
   created_unix: 1603011540 # grouped with id:7
+
+- id: 8
+  user_id: 1
+  op_type: 12 # close issue
+  act_user_id: 1
+  repo_id: 1700
```
Comment (6543) [inline models/fixtures/action.yml]:
```suggestion
  repo_id: 1700 # dangling intentional
```

**Review by 6543 — APPROVED** [review]

**Review by Gusted — APPROVED** [review]
LGTM, seems a bit hacky tbh. Seems like we should actually fix `loadRepoOwner` to not panic.

**Review by 6543 — CHANGES_REQUESTED** [review]
`TestPrivateActivityNoVisibleForPublic` fails related :/

**Comment by singuliere** [comment]
> LGTM, seems a bit hacky tbh. Seems like we should actually fix `loadRepoOwner` to not panic.

I agree. Now that the tests are passing I'll try to follow your suggestion.

**Comment by singuliere** [comment]
@Gusted Added a safeguard so **loadRepoOwner** is no longer susceptible to panic on action.Repo == nil

**Comment by techknowlogick** [comment]
@singuliere please refrain from force pushes, it makes it hard to review PRs

**Review by lunny — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by techknowlogick** [comment]
@singuliere please send backport :)

**Pull request merged**